[ASPA-004] Payment System

Dan1224444 opened the pull request:
**Issue:**
For the user to make a payment online, we must have a payment function which accepts different payment methods.

**Solution:**
In order for the user to be able to make an online payment, we must integrate a payment API. We have decided to use Stripe's card payment API for this project. Once the user clicks on the process payment button, they will be redirected to Stripe's payment page. If their payment is successful, the user will be redirected to the success page, otherwise the home page.

**Risk area:**
Currently redirects to an empty success page.
IEPay not yet implemented (Only card payment is working)

**Reviewed by:**
Lucas Gao
Ramond Feng
Will Zhang
Jiaru Lin

MartinTiangco committed:
added email model function to be called

MartinTiangco committed:
nothing changed

MartinTiangco committed:
Revert "nothing changed"

This reverts commit fae371502ac4b78100451433def7572d5267f02c.

Will-ZYS committed:
add PHPMailer files and configured mailing details, however, waiting for App password from google account

Will-ZYS committed:
changed mail password to APP password for mac, mailing function done

Commit:
PHP script for payment

Commit:
PHP script for payment

Commit:
Added redir.php

Commit:
Added redir.php

Commit:
added application/payment/stripe.php

Commit:
added application/payment/stripe.php

Will-ZYS committed:
email sample modification

Will-ZYS committed:
email sample modification

Will-ZYS committed:
prepare to update ASPA-003 branch

Will-ZYS committed:
prepare to update ASPA-003 branch

Will-ZYS committed:
email function intergrated with submit button on payment screen

Commit:
Fix file location, added constants

Commit:
Fix file location, added constants

Commit:
Added stripe.php into views

Commit:
Added stripe.php into views

Commit:
Fix public key

Commit:
Fix public key

Commit:
Adding UI button

Commit:
Adding UI button

lucas2005gao committed:
Refactored Stripe code and added payment controller logic

lucas2005gao committed:
Refactored Stripe code and added payment controller logic

Will-ZYS committed:
email functionality successfully intergrated with email input, cash/bank transfer toggle button, and submmit button

Will-ZYS committed:
Merge branch 'master' of https://github.com/lucas2005gao/ASPA-EnrollmentForm

Will-ZYS committed:
Merge branch 'master' into ASPA-003/Implementing-the-design-of-emails

Commit:
refactored code into Stripe_Model

Commit:
refactored code into Stripe_Model

Commit:
refactored stripe.php into Stripe_Model

Commit:
refactored stripe.php into Stripe_Model

Commit:
fixed minor bug in CheckPayment functin in Stripe_Model

Commit:
fixed minor bug in CheckPayment functin in Stripe_Model

Commit:
Comment code and tidy

Commit:
Comment code and tidy

Will-ZYS committed:
added comments to email model

Will-ZYS committed:
solving changes from recent other pull request

Will-ZYS committed:
solving changes from recent other pull request

Will-ZYS committed:
solving changes from recent other pull request

Will-ZYS committed:
removed sheetapi_test function in EnrollmentForm Controller

Will-ZYS committed:
removed email.php

Will-ZYS committed:
removed welcome controller

lucas2005gao commented on application/config/config.php:
```diff
@@ -23,7 +23,7 @@
 | a PHP script and you can easily do that on your own.
 |
 */
-$config['base_url'] = '';
```
Can you please revert this line of change. The base_url should be automatically detected. This may cause some issues under the developer's local development environment but let's try to stick with an empty string on the master branch.

lucas2005gao replied:
Can you please revert this line of change. The base_url should be automatically detected. This may cause some issues under the developer's local development environment but let's try to stick with an empty string on the master branch.

Commit:
Fix minor bug with config file

Commit:
Fix minor bug with config file

Commit:
Fixed base url link

Commit:
Fixed base url link

Will-ZYS committed:
Merge pull request #8 from zyan225/ASPA-003/Implementing-the-design-of-emails

[ASPA-003] Implementing the design of emails and functionality of sending emails

lin8231 reviewed (approved):
Great work Daniel!!

Will-ZYS commented:
If possible please call sendEmail method in emailModel to send confirmation email after payment successful

MartinTiangco committed:
Aspa 017/integration of email validation (#13)

* Deleted eventlistener for constant email validation

* Added a method to the controller that the ajax call in enrollmentForm.js will execute

* added ajax call

* Email Validation now working.

* [ASPA-007] Integration of Email Validation

Issue: validite() function needed updating to differentiate between error messages
Solution: updated validate() function to give 3 types of responses (success, signed up but not paid, not signed up)
Risk: fields of create json functio may need to be altered or removed
Reviewed by: Martin

* Made conditional statements for members who have paid (for membership) and signed up, members who signed up but not paid and members who haven't signed up. Different json objects are created from those statements

* added autofocus and refactored showing/hiding feedback messages (tick, warning, loading) on name/email form page into their own functions

* Added two different error warnings - has signed up but not paid - not signed up

* Changed all icons to be absolute positioning. Added a div container with relative positioning to nest the icons in. Doing this will mean that the icons now appear next to the email input field (before, the icons stack on top of each other. Even if only one icon was visible the unvisible icons still occupied space, meaning they were stacked on each other)

* show loading after name validation and removed green border when hiding all warnings

* Changed comments above function declarations to be documentation style (i.e. /** */)

* Altered the background colour of the spinner.svg to match the background of the page.

* Rebase with master complete

Co-authored-by: Jiaru <lin8231@outlook.com>

Commit:
Fixed bugs after rebase, added send email function after Stripe Payment

Commit:
Fixed more bugs for pull req
